Chat View Timestamps

mcowger opened the pull request:
## Context

Add optional timestamps to the chat view

## Implementation

Very basic, following the implementation fo the taskTimeline.  Adds option to DisplaySettings, and conditionally renders the timestamp in the chat view.

## Screenshots

<img width="323" height="435" alt="Screenshot 2025-09-19 at 2 57 42 PM" src="https://github.com/user-attachments/assets/390619dd-3c95-4209-8cd0-a73ccec81ea7" />

## How to Test

Enabled by default, so simply make a request in the chat window.   Go to display settings, and disable the feature, save, and then try again.


## Get in Touch

mcowger

mcowgeratsigma committed:
feat: add Synthetic provider support and related configurations and documentation.

adamhill commented on src/services/code-index/processors/scanner.ts:
```diff
@@ -354,26 +354,45 @@ export class DirectoryScanner implements IDirectoryScanner {
 	): Promise<void> {
-		if (batchBlocks.length === 0) return
+		if (batchBlocks.length === 0) {
+			console.debug("[DirectoryScanner] Skipping empty batch processing")
```
Wait... why is the timestamp PR modifying the CBI scanner :-)

SU-PER-NO-VA!!!!! <shakes-fist />

mcowger replied:
Yeah, thats me and git not getting along, not the model :).  I've corrected it.

mcowgeratsigma committed:
fix(webview): improve conditional model loading logic

When the API provider is not 'kilocode', we now return openRouterDefaultModelId for kilocodeDefaultModel
instead of calling getKilocodeDefaultModel.  This prevents undesirable external internet connections for users behind restrictive firewalls

billycao committed:
Update docs style

billycao committed:
Add missing translations for Synthetic

Commit:
add approvals

Commit:
add testcases

Commit:
remove two i did not like

Commit:
print better formatted

Commit:
extend examples

mcowgeratsigma committed:
Rebase onto main

kevinvandijk committed:
Remove mentions of Roo and Cline from package.json

b3nw committed:
Implement Chutes.AI as dynamic provider

Migrates Chutes.AI from static model list to dynamic fetching via /v1/models endpoint, following the pattern used by other dynamic providers like DeepInfra and OpenRouter.

Key changes:
- Add getChutesModels fetcher with flexible Zod schema for API response
- Fix price parsing: Chutes returns dollars per 1M tokens (not per token)
- Add chutes to dynamic provider list
- Add unit tests

chrarnoldus committed:
Optimistically enable caching for (new) Claude and Gemini models

chrarnoldus committed:
Implement OpenRouter ZDR

chrarnoldus committed:
Rename

chrarnoldus committed:
Translations

billycao committed:
Add kilocode_change comments

billycao committed:
Move code if there is already a kilocode_change statement

chrarnoldus committed:
Apply suggestions from code review

chrarnoldus committed:
Synthetic changeset

chrarnoldus committed:
Make timeout for Ollama and LM Studio configurable

chrarnoldus committed:
Extract const

chrarnoldus committed:
Improve error display

chrarnoldus committed:
Changeset

chrarnoldus committed:
Add mocks

chrarnoldus committed:
ZDR changeset

chrarnoldus committed:
Remove Switch to Kilo Code button

chrarnoldus committed:
Remove unused strings

chrarnoldus committed:
Update docs

chrarnoldus committed:
Add Provider Routing docs for Kilo Code and OpenRouter

EamonNerbonne committed:
Simplify runner very slightly

EamonNerbonne committed:
Fix type errors

chrarnoldus committed:
Add Revert All Changes button

chrarnoldus committed:
Translations

EamonNerbonne committed:
Remove nasty bug checkbox

chrarnoldus committed:
Extract component and add confirm buttons

chrarnoldus committed:
Changeset

EamonNerbonne committed:
grinding gears

kevinvandijk reviewed (changes requested):
Looks good overall. 2 minor comments and one general thing: the alignment seems slightly off. Can you change that to be middle aligned with eachother? 
<img width="1990" height="170" alt="image" src="https://github.com/user-attachments/assets/0a131868-ecd1-4ff0-be3e-b4423442cd62" />

chrarnoldus committed:
Add kilocode_change markers

chrarnoldus committed:
Mostly revert this file to Roo's version

chrarnoldus committed:
Add kilocode_change comments

chrarnoldus committed:
Add kilocode_change comments

chrarnoldus committed:
Changeset

mcowger committed:
Update .changeset/blue-suits-peel.md

Co-authored-by: Kevin van Dijk <kevin@sqre.one>

mcowgeratsigma committed:
Convert timestamps to element.

mcowger committed:
Merge branch 'main' into mcowger/timestamps

mcowgeratsigma committed:
Fixed vertical alignment

kevinvandijk committed:
Merge branch 'main' into mcowger/timestamps

kevinvandijk committed:
Merge branch 'main' into mcowger/timestamps

kevinvandijk committed:
Add translations

kevinvandijk committed:
Add kilocode markers

kevinvandijk reviewed (approved):
Added some small fixes and resolved some merge conflicts, looks good now!
